src/components/item/form/EditModalWrapper.tsx

```diff
@@ -0,0 +1,148 @@
+import { ComponentType as CT, Dispatch, useState } from 'react';
+import { toast } from 'react-toastify';
+
+import { Button } from '@mui/material';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+
+import { routines } from '@graasp/query-client';
+import { DiscriminatedItem } from '@graasp/sdk';
+import { BUILDER, COMMON, FAILURE_MESSAGES } from '@graasp/translations';
+
+import CancelButton from '@/components/common/CancelButton';
+import { DOUBLE_CLICK_DELAY_MS } from '@/config/constants';
+import { useBuilderTranslation, useCommonTranslation } from '@/config/i18n';
+import notifier from '@/config/notifier';
+import { mutations } from '@/config/queryClient';
+import {
+  EDIT_ITEM_MODAL_CANCEL_BUTTON_ID,
+  EDIT_MODAL_ID,
+  ITEM_FORM_CONFIRM_BUTTON_ID,
+} from '@/config/selectors';
+import { isItemValid } from '@/utils/item';
+
+const { editItemRoutine } = routines;
+
+export interface EditModalContentPropType {
+  item?: DiscriminatedItem;
+  setChanges: (payload: Partial<DiscriminatedItem>) => void;
+  updatedProperties: Partial<DiscriminatedItem>;
+}
+
+export type EditModalContentType = CT<EditModalContentPropType>;
+
+type Props = {
+  ComponentType: EditModalContentType;
```

Comment:
I understand the idea, maybe using a more specific name for that property would help understand it better? How about `FormComponent` ?